pkgs/by-name/sp/spectral-language-server/package.nix

```diff
@@ -0,0 +1,93 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, typescript
+, jq
+}:
+let
+  # Instead of the build script that spectral-language-server provides (ref: https://github.com/luizcorreia/spectral-language-server/blob/master/script/vscode-spectral-build.sh), we build vscode-spectral manually.
+  vscodeSpectral = buildNpmPackage rec {
+    pname = "vscode-spectral";
+    version = "1.1.2";
+
+    src = fetchFromGitHub {
+      owner = "stoplightio";
+      repo = "vscode-spectral";
+      rev = "v${version}";
+      sha256 = "sha256-TWy+bC6qhTKDY874ORTBbvCIH8ycpmBiU8GLYxBIiAs=";
+    };
+
+    npmDepsHash = "sha256-mMVDhdYAEmbW/jaK6q4YhcGbE10mfrvlr8CNCFBMpIU=";
+
+    npmFlags = [ "--ignore-scripts" ];
+
+    postPatch = ''
+      cp ${./vscode-spectral-package-lock.json} ./package-lock.json
+      chmod +w package-lock.json
+    '';
+
+    dontNpmBuild = true;
+
+    nativeBuildInputs = [ typescript jq ];
+
+    buildPhase = ''
+      runHook preBuild
+      cp server/tsconfig.json server/tsconfig.json.bak
+      jq '.compilerOptions += {"module": "NodeNext", "moduleResolution": "NodeNext"}' server/tsconfig.json.bak > server/tsconfig.json
+
+      # FIXME: vscode-spactral depends on @rollup/pluginutils, but it may have a bug that doesn't provide the type definitions for NodeNext module resolution. (ref: https://github.com/rollup/plugins/issues/1192)
+      # tsc detects some type errors in the code. However we ignore this because it's not a problem for the final build if server/dist is generated.
+      tsc -p server || true
+      test -d server/dist
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      mkdir -p $out
+      cp -R server/dist $out
+      runHook postInstall
+    '';
+  };
+in
+buildNpmPackage rec {
+  pname = "spectral-language-server";
+  version = "1.0.8";
+
+  src = fetchFromGitHub {
+    owner = "luizcorreia";
+    repo = pname;
+    rev = "c9a7752b08e6bba937ef4f5435902afd41b6957f";
+    hash = "sha256-VD2aAzlCnJ6mxPUSbNRfMOlslM8kLPqrAI2ah6sX9cU=";
+  };
+
+  npmDepsHash = "sha256-/SvVDX3CSPKs6FVANVzXcrxhXrpBA2WE2bGdg5yYjK0=";
+
+  postPatch = ''
+    cp ${./package-lock.json} ./package-lock.json
```

Comment:
`npm install` doesn't work and throws errors with this lockfile because it doesn't contain integrity and resolve fields. I tried to add missing properties with jq, but there aren't fields of almost packages, so I feel so hard to add manually.